packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -104,266 +68,583 @@ describe("Garbage Collection Tests", () => {
+            assert(
+                mockLogger.matchEvents([
+                    { eventName: inactiveObjectRevivedEvent, deleteTimeoutMs, inactiveNodeId: nodes[3] },
+                ]),
+                "inactiveObjectRevived event not generated as expected",
+            );
+        });
 
-        // Remove node 2's reference from node 1. This should make node 2 and node 3 unreferenced.
-        defaultGCData.gcNodes[nodes[1]] = [];
+        it("generates inactive events once per node", async () => {
+            const garbageCollector = createGarbageCollector();
 
-        await garbageCollector.collectGarbage({ runGC: true });
+            // Remove node 3's reference from node 2.
+            defaultGCData.gcNodes[nodes[2]] = [];
 
-        // Change all nodes.
-        changeAllNodes(garbageCollector);
+            await garbageCollector.collectGarbage({ runGC: true });
 
-        // Validate that no inactive events are generated yet.
-        validateNoInactiveEvents();
+            await waitForDeleteTimeout();
 
-        // Wait for unreferenced timer (if any) to expire.
-        await waitForDeleteTimeout();
+            // Change all nodes. This should result in an inactiveObjectChanged event for node 2 and node 3 since they
+            // are inactive.
+            changeAllNodes(garbageCollector);
+            assert(
+                mockLogger.matchEvents([
+                    { eventName: inactiveObjectChangedEvent, deleteTimeoutMs, inactiveNodeId: nodes[3] },
+                ]),
+                "inactiveObjectChanged event not generated as expected",
+            );
 
-        // Change all nodes. This should result in an inactiveObjectChanged event for node 2 and node 3 since they
-        // are inactive.
-        changeAllNodes(garbageCollector);
-        assert(
-            mockLogger.matchEvents([
-                { eventName: inactiveObjectChangedEvent, deleteTimeoutMs, inactiveNodeId: nodes[2] },
-                { eventName: inactiveObjectChangedEvent, deleteTimeoutMs, inactiveNodeId: nodes[3] },
-            ]),
-            "inactiveObjectChanged event not generated as expected",
-        );
+            // Change all nodes. There shouldn't be any more inactive events since for each node the event is only
+            // once.
+            changeAllNodes(garbageCollector);
+            validateNoInactiveEvents();
+        });
 
-        // Add reference to node 3 from node 1.
-        defaultGCData.gcNodes[nodes[1]] = [ nodes[3] ];
+        it("generates inactive events for nodes that are inactive on load", async () => {
+            // Create GC state where node 3's unreferenced time was > deleteTimeoutMs ago.
+            // This means this node should become inactive as soon as its data is loaded.
+
+            // Create a snapshot tree to be used as the GC snapshot tree.
+            const gcSnapshotTree = getDummySnapshotTree();
+            const gcBlobId = "gc_blob";
```

Comment:
Should this value be in sync with some value in code? Or it can be any value? I can't find it in code otherwise.

Reply:
This is the the blob id that is generated when a blob is uploaded. The snapshot tree that is downloaded has a mapping of "blob key" to "blob id". The key here is anything that starts with "gcBlobPrefix" which matches the code.
Added a comment to clarify some of it.